channel/allocation.go

```diff
@@ -603,6 +638,19 @@ func (s *SubAlloc) BalancesEqual(b []Bal) bool {
 	return true
 }
 
+// indexMapEqual returns whether s.indexMap equals b.
```

Comment:
No. The order does matter. `subPart[indexMap[i]] == part[i]`

Reply:
Yes sure the order matters, i meant to check for a valid permutation in the mathematical sense.  
Otherwise there could be out of range values, but whatever.